### PR TITLE
Fix clippy lints

### PR DESCRIPTION
**Pull request by baumanj**
Do all the fixes for the core repo to get to parity with the lints the habitat repo allows/denies. Fixes for the different lints are split up by commit for ease of review.

Once this merges, the following `habitat` and `builder` PRs will be necessary when pulling in these changes:
- https://github.com/habitat-sh/habitat/pull/6288
- https://github.com/habitat-sh/builder/pull/973

### CODE REVIEW

**Comment by thesentinels** [comment]
Thanks for the pull request! Here is what will happen next:
 1. Your PR will be reviewed by the maintainers
 2. If everything looks good, one of them will `approve` it, and your PR will be merged.

Thank you for contributing!

File: Makefile [hunk 1]
```diff
@@ -58,91 +51,85 @@ endef
...
+               clippy::wrong_self_convention \
+               renamed_and_removed_lints
+
+lint:
+	$(run) cargo clippy --all-targets --tests $(CARGO_FLAGS) -- \
+					$(addprefix -A ,$(UNEXAMINED_LINTS)) \
+					$(addprefix -A ,$(ALLOWED_LINTS)) \
+					$(addprefix -W ,$(LINTS_TO_FIX)) \
+					$(addprefix -D ,$(DENIED_LINTS))
+.PHONY: lint-
```
Comment (raskchanky) [inline Makefile]: Should this be `.PHONY: lint`?

Comment (baumanj) [reply Makefile]: Yes! Good catch.

File: Makefile [hunk 2]
```diff
@@ -58,91 +51,85 @@ endef
...
+               clippy::wrong_self_convention \
+               renamed_and_removed_lints
+
+lint:
+	$(run) cargo clippy --all-targets --tests $(CARGO_FLAGS) -- \
+					$(addprefix -A ,$(UNEXAMINED_LINTS)) \
+					$(addprefix -A ,$(ALLOWED_LINTS)) \
+					$(addprefix -W ,$(LINTS_TO_FIX)) \
+					$(addprefix -D ,$(DENIED_LINTS))
+.PHONY: lint
```
Comment (raskchanky) [inline Makefile]: I ❤️ this kind of cleanup to the `Makefile` btw.  It makes the readability so much better.

**Pull request merged**